Document that the sd-ssh VM needs more space

I think due to the change to use Docker, the root OS disk of sd-ssh fills up a bit faster than it used to. The last couple of commits that triggered CI on my setup, failed during either `make clone` or `make dev` for this reason:

```
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0 Error Summary
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0 -------------
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0 Disk Requirements:
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0    At least 300MB more space needed on the / filesystem.
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 110.0 make: *** [Makefile:43: install-deps] Error 1
May 31 05:14:28 sd-ssh qubes.VMShell-dom0[2443]: #8 ERROR: process "/bin/sh -c make install-deps" did not complete successfully: exit code: 2
```

I adjusted the sd-ssh VM's root disk to be 40GB instead of 10GB.. we should probably document it, and we may want to add a `docker system prune --force` to the end of our runner.py.

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -1,0 +1,148 @@
+# Installation instructions
+
+This document explains how to install the CI for Securedrop Workstation.
+
+It involves a combination of dom0 and VM configuration on a Qubes installation, as well as steps in
+Github/Tailscale.
+
+# Qubes install and initial provisioning
+
+1. Download and verify Qubes per
+   [these instructions](https://workstation.securedrop.org/en/stable/admin/install.html#download-and-verify-qubes-os).
+
+2. Install Qubes, mostly following
+   [these instructions](https://workstation.securedrop.org/en/stable/admin/install.html#install-qubes-os-estimated-wait-time-30-45-minutes)
+   (see exceptions below).
+
+2a. Keep all configuration defaults, except:
+
+- Turn off FDE (we could later investigate doing this with FDE enabled, see
+  [this issue](https://github.com/freedomofpress/securedrop/issues/816))
+- Uncheck the creation of the personal and work qubes (this will show up as an option after reboot)
+
+2b. In dom0, confirm you can start sys-usb (because it’s given us some issues with different
+hardware) by running `qvm-start sys-usb`.
+
+2c. Update dom0 per
+[these instructions](https://workstation.securedrop.org/en/stable/admin/install.html#apply-dom0-updates-estimated-wait-time-15-30-minutes)
+
+2d. Run any updates you see in the Qubes menu and then reboot.
+
+3. In dom0, install `make` and then create the sd-ssh StandaloneVM:
+
+```
+sudo qubes-dom0-update make
+sudo qvm-create --standalone --template fedora-37 --label red sd-ssh
+qvm-volume resize sd-ssh:root 50G
+qvm-volume resize sd-ssh:private 20G
+qvm-tags sd-ssh add sd-client
+```
+
+Also ensure that you check the box to 'Start qube automatically on boot' in the Qubes settings.
+
+# Install dependencies on sd-ssh VM
+
+Open a terminal in the sd-ssh VM and perform the following steps to install the core dependencies
+and Tailscale.
+
+```
+sudo dnf install openssh-server rpm-build dnf-plugins-core python3-pip python3-flask python3-paramiko python3-scp
+sudo pip3 install github-webhook
+sudo systemctl ssh enable
+
+curl -fsSL https://tailscale.com/install.sh | sh
+sudo tailscale up --advertise-tags=tag:servers,tag:sd-ci-servers
+```
+
+You will need to go and complete the approval of the device in Tailscale as an admin, by copying the
+link that is returned in the last step.
+
+Then come back and proceed with the following steps.
+
+```
+sudo -i
+
+iptables -I INPUT 3 -m tcp -p tcp --dport 22 -i tailscale0 -j ACCEPT
+ip6tables -I INPUT 3 -m tcp -p tcp --dport 22 -i tailscale0 -j ACCEPT
+iptables -I INPUT 3 -m tcp -p tcp --dport 5000 -i tailscale0 -j ACCEPT
+ip6tables -I INPUT 3 -m tcp -p tcp --dport 5000 -i tailscale0 -j ACCEPT
+iptables-save > /etc/qubes/iptables.rules
+ip6tables-save > /etc/qubes/ip6tables.rules
+
+sudo dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
+sudo dnf install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+sudo usermod -a -G docker user
+systemctl enable docker
+```
+
+# Install the CI scripts from this repository
+
+You're nearly done! Now you need to install the actual CI scripts, systemd unit files, and other
+config from this very repo into your dom0 and sd-ssh.
+
+## Instructions for dom0
+
+1. Put the file `runner.py` in `/home/user/`. Ensure it is executable.
+
+2. Put the files `qubes.SDCIRunner` and `qubes.SDCICanceler` in `/etc/qubes-rpc/`. Ensure they are
+   executable.
+
+3. Put the files `qubes.SDCIRunner.policy` and `qubes.SDCICanceler.policy` as
+   `/etc/qubes-rpc/policy/qubes.SDCIRunner` and `/etc/qubes-rpc/policy/qubes.SDCICanceler`.
+
+These policy files do not need to be executable.
+
+4. Put the files `sd-ssh-update.service` and `sd-ssh-update.timer` in `/etc/systemd/system/` and run
+   `systemctl enable sd-ssh-update.service; systemctl enable sd-ssh-update.timer; systemctl start sd-ssh-update.timer`
+   as root.
+
+5. Put the file `sd-ssh-update.sh` in `/home/user/`.
+
+## Instructions for sd-ssh
+
+1. Create the CI runner working directory with `sudo mkdir /var/lib/sdci-ci-runner`.
+
+2. Put the file `webhook.py` in `/var/lib/sdci-ci-runner`.
+
+3. Create a `config.json` copied from
+   [this example](https://github.com/freedomofpress/securedrop-workstation/blob/main/files/config.json.example)
+   and a `sd-journalist.sec` copied from
+   [this example](https://github.com/freedomofpress/securedrop-workstation/blob/main/sd-journalist.sec.example),
+   and store them in `/var/lib/sdci-ci-runner`.
+
+4. At this point, `sudo chown -R user.user /var/lib/sdci-ci-runner`.
+
+5. Put `upload-report` and `cancel.py` in `/home/user/bin`.
+
+6. Generate a PAT in Github with full `repo:` access and ensure that that PAT is set in the
+   `upload-report` script as the `github_token` variable, so that the script can post git commit
+   statuses back to Github.
+
+7. Put `sdci-repo-webhook.service` in `/etc/systemd/system/`. Set a value for
+   `SDCI_REPO_WEBHOOK_SECRET` in this file. Also adjust the `FLASK_RUN_HOST` to the IP of your
+   sd-ssh machine's Tailscale IP so that the service listens only on that interface.
+
+8. Enable and start the flask webhook service via systemd:
+   `sudo systemctl daemon-reload; sudo systemctl enable sdci-repo-webhook; sudo systemctl start sdci-repo-webhook`.
+
+9. Configure the webhook in your repository for the 'push' event, with the same secret you put in
+   the systemd file in step 7.
+
+The Payload URL of the webhook should be https://ws-ci-runner.securedrop.org/hook/postreceive and
+the Content type should be `application/json`. Ensure you keep `Enable SSL verification` turned on.
+
+10. Generate an SSH key on sd-ssh with `ssh-keygen -t ed25519 -f ~/.ssh/id_ed25519_sdci_upload`, and
+    ensure that this key is in the `/home/wscirunner/.ssh/authorized_keys` on the tailscale proxy
+    droplet. This ensures that the `upload-report` script can successfully scp up the log file to
+    the proxy droplet. Run
+    `ssh -i ~/.ssh/id_ed25519_sdci_upload wscirunner@ws-ci-runner.securedrop.org` once, on the
+    sd-ssh VM, to accept the host key signature for the first time.
+
+## Reboot and test
+
+Do a full reboot of the Qubes system.
+
+Then, double-check that sd-ssh has started automatically at boot and that it has started the Flask
+webhook service (look for a process `/usr/bin/python3 -m flask run`).
+
+After that, you can push a commit to the repository and test the webhook and CI process works.

--- a/README.md
+++ b/README.md
@@ -2,86 +2,77 @@
 
 ## About
 
-This collection of scripts is for running the securedrop-workstation CI on a Qubes machine.
+This collection of scripts is for running the securedrop-workstation CI on a Qubes device.
 
-Given the nature of Qubes, it is a complex component with a few moving parts.
+## Installation instructions
 
-This document tries to help FPF/SD engineers install the scripts on a Qubes 4.1+ machine.
-
-## Instructions for dom0
-
-1. Set up Qubes 4.1 per the Nextcloud doc.
-
-2. Put the file `runner.py` in `/home/user/`.
-
-3. Put the files `qubes.SDCIRunner` and `qubes.SDCICanceler` in `/etc/qubes-rpc/`. Ensure they are executable.
-
-4. Put the files `qubes.SDCIRunner.policy` and `qubes.SDCICanceler.policy` as `/etc/qubes-rpc/policy/qubes.SDCIRunner` and `/etc/qubes-rpc/policy/qubes.SDCICanceler`.
-
-
-## Instructions for sd-ssh
-
-1. `sudo mkdir /var/lib/sdci-ci-runner`
-
-2. Put the file webhook.py in `/var/lib/sdci-ci-runner` . You may need to change the IP of the binding IP to your sd-ssh machine's tailscale IP
-
-3. Put a `config.json` and `sd-journalist.sec` (prepared earlier as part of installation instructions) in `/var/lib/sdci-ci-runner`. At this point, `sudo chown -R user.user /var/lib/sdci-ci-runner`
-
-4. Put `upload-report` and `cancel.py` in `/home/user/bin`. Generate a PAT in Github with full `repo:` access and ensure that that PAT is in `upload-report`, so that the script can post git commit statuses back to Github.
-
-5. Put `sdci-repo-webhook.service` in `/etc/systemd/system/`. Configure a webhook 'secret' in this systemd file. Also adjust the `FLASK_RUN_HOST` to the IP of your sd-ssh machine's Tailscale IP.
-
-6. Install some dependencies: `sudo dnf install python3-pip python3-flask python3-paramiko python3-scp; sudo pip3 install github-webhook`
-
-7. Enable and start the flask service via systemd: `systemctl daemon-reload; systemctl enable sdci-repo-webhook; systemctl start sdci-repo-webhook`
-
-8. You will need to have opened port 5000 for the webhook in your tailscale iptables rules (`iptables -I INPUT 3 -m tcp -p tcp --dport 5000 -i tailscale0 -j ACCEPT`), and configured relevant Tailscale ACLs in the Tailscale admin dashboard. Riley took care of the latter for Mig.
-
-9. Configure the webhook in your repository for the 'push' event, with the same secret you put in the systemd file in step 5. The URL would be https://ws-ci-runner.securedrop.org/hook/postreceive per https://github.com/freedomofpress/infrastructure/pull/4111 . You will also need to have an Nginx proxy somewhere answering for the 'outer' HTTPS request, and proxying through to your sd-ssh machine's port 5000 Flask app (via Tailscale)
-
-10. Generate an SSH key on sd-ssh with `ssh-keygen -t ed25519 -f ~/.ssh/id_ed25519_sdci_upload`, and ensure this key is in the `/home/wscirunner/.ssh/authorized_keys` on the ws-ci-runner droplet proxy (so that `upload-report` can scp up the log results). Run `ssh -i ~/.ssh/id_ed25519_sdci_upload wscirunner@ws-ci-runner.securedrop.org` once to accept the host key signature.
-
-11. Ensure you have Docker installed on the sd-ssh appVM, and that the 'user' user is a member of the docker group. This is needed for the RPM build process of SD Workstation.
-
-
-## Test it
-
-Try and push a commit and see if the webhook works!
-
+Please see the [INSTALL.md](INSTALL.md).
 
 ## How it works
 
-1. When the webhook delivers the payload to the endpoint, the webhook obtains info about the commit/repo and clones the repo into `/var/lib/sdci-ci-runner/securedrop-workstation_{SHA}` on the sd-ssh VM.
+@TODO add architecture diagram.
 
-2. It will then trigger an RPC call to dom0 to run the `runner.py` (wrapped in flock to avoid concurrent builds).
+1. The webhook in Github delivers the payload to the proxy droplet.
 
-3. The runner.py reports a commit status back to Github that the build has started.
+2. The proxy droplet proxies the payload through to the sd-ssh VM running on a Qubes device, via the
+   Tailscale tunnel.
 
-4. The runner.py tarballs up the codebase from sd-ssh and proceeds with the `make clone; make dev; make test` sequence, logging to a log file the whole time.
+3. The sd-ssh VM sends a commit status to Github notifying that the build has been received and is
+   pending.
 
-5. The runner.py then leverages the securedrop-workstation's `scripts/sdw-admin.py --uninstall --force` to tear everything down, along with cleaning up some remaining cruft.
+4. The sd-ssh VM clones the repository and checks out the commit learned from the payload, at
+   `/var/lib/sdci-ci-runner/securedrop-workstation_{SHA}`.
 
-The runner.py will detect if any of the commands succeed or fail but it should not abort on failure (so that the teardown still completes).
+5. The sd-ssh VM sends an RPC call to the dom0 to run the `runner.py` (wrapped in flock to avoid
+   concurrent builds).
 
-6. At the end of the process, the dom0 will copy its log file to the sd-ssh machine and then call `upload-report` via `qvm-run` on that machine with the status of the build.
+6. The runner.py reports a commit status back to Github (via sd-ssh) that the build has started (or,
+   if there is a build already running, that it is queued).
 
-That script will upload the log to the ws-ci-runner proxy for viewing in a browser at https://ws-ci-runner.securedrop.org, and will also post a commit status to Github, with the `target_url` pointing to the HTTPS URL of that log file on the ws-ci-runner, and with the status of the build.
+7. The runner.py tarballs up the codebase from the sd-ssh VM, and proceeds with the
+   `make clone; make dev; make test` sequence, logging all output to a log file.
 
+8. The runner.py then leverages the securedrop-workstation's `sdw-admin.py --uninstall --force` to
+   tear everything down, along with cleaning up some remaining cruft.
+
+The runner.py will detect if any of the commands succeed or fail but it should not abort on failure
+(so that the teardown still completes).
+
+9. At the end of the process, the dom0 will copy its log file to the sd-ssh VM and then calls
+   `upload-report` on the sd-ssh VM with the status of the build.
+
+10. That `upload-report` script will upload the log to the ws-ci-runner proxy for viewing in a
+   browser at https://ws-ci-runner.securedrop.org, and will also post a commit status to Github,
+   with the `target_url` pointing to the HTTPS URL of that log file on the ws-ci-runner, and with
+   the status of the build. At this point, the commit shows either a green tick or a red cross and
+   has a link to the log file.
 
 ## Queuing and canceling builds
 
-The webhook can handle multiple commits delivered to it. The jobs get issued to the dom0 with a maximum `flock` wait of 86400s (24h).
+The webhook can handle multiple commits delivered to it. The jobs get issued to the dom0 with a
+maximum `flock` wait of 86400s (24h).
 
-If another job is already running, it means the lock is held, so the other jobs wait for the lock to be released before starting.
+If another job is already running, it means the lock is held, so the other jobs wait for the lock to
+be released before starting.
 
 Once the lock releases, one of the pending jobs will claim the lock and start running.
 
-While a job is waiting, the commit in Github has a status of 'pending' with the message 'The build is queued'.
+While a job is waiting, the commit in Github has a status of 'pending' with the message 'The build
+is queued'.
 
-When a build starts, the commit status changes to a description of 'The build is running'. The commit status state is technically still 'pending' because Github makes no distinction between 'queued' and 'running', except in the description field of the commit status.
+When a build starts, the commit status changes to a description of 'The build is running'. The
+commit status state is technically still 'pending' because Github makes no distinction between
+'queued' and 'running', except in the description field of the commit status.
 
-If you need to cancel a build that is queued, run `cancel.py --sha xxxxxxxx` on the sd-ssh VM. This will:
+If you need to cancel a build that is queued, run `cancel.py --sha xxxxxxxx` on the sd-ssh VM. This
+will:
 
- * remove the codebase that was checked out to this commit on sd-ssh
- * kill the pending process on the dom0 (by way of the qubes.SDCICanceler RPC script)
- * update the git commit status at Github to say that this build was canceled by an administrator. The commit status will now be of state 'error' with a red cross.
+- remove the codebase that was checked out to this commit on sd-ssh
+- kill the pending process on the dom0 (by way of the qubes.SDCICanceler RPC script)
+- update the git commit status at Github to say that this build was canceled by an administrator.
+  The commit status will now be of state 'error' with a red cross.
+
+## Automatic updates of the sd-ssh VM
+
+The installation adds a systemd timer and script to perform daily updates of the sd-ssh VM to keep
+it up to date.

